src/databricks/sql/auth/thrift_http_client.py

```diff
@@ -191,7 +189,7 @@ def flush(self):
 
         # Saves the cookie sent by the server response
         if "Set-Cookie" in self.headers:
-            self.setCustomHeaders(dict("Cookie", self.headers["Set-Cookie"]))
+            self.setCustomHeaders({"Cookie", self.headers["Set-Cookie"]})
```

Comment:
I don't think setCustomHeaders is correct here, this will replace all existing headers . Shouldn't it be something like headers.update to append ?